Sources/LocheckCommand/main.swift

```diff
@@ -42,43 +42,54 @@ private func withProblemReporter(ignore: [String], _ block: (ProblemReporter) ->
     print("Finished validating")
 }
 
-private let ignoreHelpText: ArgumentHelp =
-    "Ignore a rule completely."
+private let ignoreHelpText: ArgumentHelp = "Ignore a rule completely."
 
 private let ignoreMissingHelpText: ArgumentHelp =
     "Ignore 'missing string' errors. Shorthand for '--ignore key_missing_from_base --ignore key_missing_from_translation'."
 
-struct XCStrings: ParsableCommand {
+private let ignoreWarningsHelpText: ArgumentHelp = "Ignore all warning-level issues."
+
+private protocol HasIgnoreWithShorthand {
+    var ignore: [String] { get }
+    var ignoreMissing: Bool { get }
+}
+
+private extension HasIgnoreWithShorthand {
+    var ignoreWithShorthand: [String] {
+        ignore + (ignoreMissing ? ["key_missing_from_base", "key_missing_from_translation"] : [])
+    }
+}
+
+struct XCStrings: HasIgnoreWithShorthand, ParsableCommand {
     static let configuration = CommandConfiguration(
         commandName: "xcstrings",
         abstract: "Directly compare .strings files")
 
     @Argument(help: "A list of .strings files, starting with the primary language")
-    private var stringsFiles: [FileArg]
+    private var stringsFiles: [FileArg] = []
 
     @Option(help: ignoreHelpText)
-    private var ignore = [String]()
+    fileprivate var ignore = [String]()
```

Comment:
I tried and failed to make a superclass for these commands. The only thing I was able to accomplish was to move `ignoreWithShorthand` to a protocol extension, requiring these properties to be fileprivate instead of private.